src/components/Link/Link.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { forwardRef, useRef, RefObject } from 'react';
+import FocusRing from '../FocusRing';
+import { useLink } from '@react-aria/link';
+import { DEFAULTS, STYLE } from './Link.constants';
+import { Props } from './Link.types';
+import './Link.style.scss';
+import classnames from 'classnames';
+import Icon from '../Icon';
+import Tooltip from '../Tooltip';
+
+const Link = forwardRef((props: Props, providedRef: RefObject<HTMLAnchorElement>) => {
+  const {
+    className,
+    title,
+    hasExternalLinkIcon,
+    externalLinkIconProps,
+    tooltipContent,
+    tooltipType,
+    disabled,
+    inverted,
+    style,
+    ...otherProps
+  } = props;
+  const internalRef = useRef();
+  const ref = providedRef || internalRef;
+
+  const mutatedProps = {
+    ...otherProps,
+    title,
+    isDisabled: disabled,
+    isInverted: inverted,
+  };
+
+  // prevent change the window.opener.location to some phishing page
+  const isExternalLink = props.target === '_blank';
+
+  const { linkProps } = useLink({ ...mutatedProps, elementType: 'a' }, ref);
+
+  let isShowIcon = false;
+
+  if (hasExternalLinkIcon === undefined && props.target === '_blank') {
+    isShowIcon = true;
+  } else if (hasExternalLinkIcon) {
+    isShowIcon = true;
+  }
```

Comment:
nitpicking: this could be simplified like this:

```suggestion
  let isShowIcon = hasExternalLinkIcon === true || isExternalLink;
```

Reply:
In this simplified change, `hasExternalLinkIcon` can't force hide it.
If `hasExternalLinkIcon  = false` but `props.target === '_blank'`, it will display icon.
So, I think we can't simple like this. I will replace `props.target === '_blank'` to `isExternalLink `

Reply:
you are right, I missed that case